plenum/server/replica.py

```diff
@@ -1967,9 +1968,10 @@ def can_pp_seq_no_be_in_view(self, view_no, pp_seq_no):
             view_no < self.viewNo and self.last_prepared_before_view_change and compare_3PC_keys(
                 (view_no, pp_seq_no), self.last_prepared_before_view_change) >= 0)
 
-    def _request_missing_three_phase_messages(self, frm: int, to: int, view_no: int) -> None:
-        for i in range(1, to - frm):
-                request_data = (view_no, frm + i)
+    def _request_missing_three_phase_messages(self, seq_frm: int, seq_to: int, view_frm: int, view_to: int) -> None:
+        for pp_seq_no in range(seq_frm + 1, seq_to + 1):
+            for view_no in range(view_frm, view_to + 1):
+                request_data = (view_no, pp_seq_no)
```

Comment:
Looks like we iterate pp_seq in range [frm+1, to], while it was [frm+1, to-1] before. Actually the current way looks more correct, but anyway, let's double-check.

Reply:
I guess it was a mistake before